.openapidoc/index.js

```diff
@@ -0,0 +1,167 @@
+const fs = require("fs");
+const yaml = require("js-yaml");
+const { https } = require("follow-redirects");
+const ghpages = require("gh-pages");
+
+// constants
+const repo = "git@github.com:PegaSysEng/eth2signer.git";
+const branch = "gh-pages";
+const spec = "../core/build/resources/main/openapi/eth2signer.yaml";
+const versionsFileUrl =
+  "https://github.com/PegaSysEng/eth2signer/raw/gh-pages/versions.json";
+const distDir = "./dist";
+const versionsFile = distDir + "/versions.json";
+
+/**
+ * Downloads the given source URL and return it as a string
+ * @param source
+ */
+async function downloadAsString(source) {
+  return new Promise((resolve, reject) => {
+    let output = "";
+    return https
+      .get(source, (response) => {
+        const { statusCode, headers } = response;
+        if (statusCode >= 400)
+          return reject(
+            new Error(
+              `Failed to download file "${source}". Status: ${statusCode}`
+            )
+          );
+
+        response.on("data", (chunk) => (output += chunk));
```

Comment:
Yeah, definitely just use `fetch` - it'll invalidate the need for this entire function.

Reply:
👍